src/lib.rs

```diff
@@ -212,6 +212,21 @@
 //! # fn main() {}
 //! ```
 #![deny(clippy::unimplemented, clippy::unwrap_used, clippy::ok_expect)]
+#![warn(
+    clippy::alloc_instead_of_core,
+    clippy::std_instead_of_alloc,
+    clippy::std_instead_of_core
+)]
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[macro_use]
+extern crate alloc;
+
+#[cfg(all(not(feature = "std"), feature = "visualizer"))]
+compile_error!("Cannot enable \"visualizer\" feature in \"no_std\" environment.");
```

Comment:
```suggestion
compile_error!("Cannot enable `visualizer` feature in `no_std` environment.");
```

Let's match these to the below.